cmd/snap/cmd_list.go

```diff
@@ -121,9 +135,68 @@ func listSnaps(names []string, all bool) error {
 	}
 	sort.Sort(snapsByName(snaps))
 
-	w := tabWriter()
-	defer w.Flush()
+	switch format {
+	case "":
+		return outputSnapsDefault(w, snaps)
+	default:
+		return outputSnapsWithFormat(w, snaps, format)
+	}
+}
+
+type fieldDesc struct {
+	name string
+	help string
+}
+
+func clientSnapFields() []fieldDesc {
+	v := reflect.TypeOf(client.Snap{})
+	n := v.NumField()
+	fields := make([]fieldDesc, n)
+	for i := 0; i < n; i++ {
+		field := v.Field(i)
+		fields[i].name = field.Name
+		fields[i].help = field.Tag.Get("help")
+	}
+
+	return fields
+}
+
+func describeListFormat(w io.Writer, format string) error {
+	fmt.Fprintf(w, `Format uses a simple template system.
+
+Use --format="{{.Name}} {{.Version}}" to get started.
+
+All elements available for snaps are:
```

Comment:
`All the elements ..`

Reply:
Yay, thanks for all the excellent coments. Updated.